pkg/controller/mpi_job_controller.go

```diff
@@ -339,7 +339,7 @@ func NewMPIJobControllerWithClock(
 		priorityClassSynced: priorityClassSynced,
 		mpiJobLister:        mpiJobInformer.Lister(),
 		mpiJobSynced:        mpiJobInformer.Informer().HasSynced,
-		queue:               workqueue.NewRateLimitingQueueWithConfig(workqueue.DefaultControllerRateLimiter(), workqueue.RateLimitingQueueConfig{Name: "MPIJobs"}),
+		queue:               workqueue.NewTypedRateLimitingQueueWithConfig(workqueue.DefaultTypedControllerRateLimiter[any](), workqueue.TypedRateLimitingQueueConfig[any]{Name: "MPIJob"}),
```

Comment:
I guess that we need to do this:

```suggestion
		queue:               workqueue.NewTypedRateLimitingQueueWithConfig(workqueue.DefaultTypedControllerRateLimiter[string](), workqueue.TypedRateLimitingQueueConfig[string]{Name: "MPIJob"}),
```

and then,

https://github.com/kubeflow/mpi-operator/blob/dbcbfae1f44d0dd9c4f99ed85908f24f19818805/pkg/controller/mpi_job_controller.go#L250

```golang
workqueue.TypedRateLimitingInterface[string]
````

Reply:
Modified my comment related to types.

Reply:
yeah I was about to comment that `any` is required.

Reply:
@ArangoGutierrez Could you address the Line 250 as I mentioned above?

Reply:
I did